### DISCUSSION TITLE
I need help or guidance

### DISCUSSION THREAD

**Discussion by fguardia** [GerryDazoo/Slinger#81] [Q&A]
I need guidance because i do not really understand what i have to do for starters this is my setup
I have a Slingbox 500 located on another country and dont know if i have to be in the same network to set it up
And do not really know where to start
I already got the info for my password but do not know where to go from here

**Comment by GerryDazoo** [comment]
You'll need the IP address and port number to use to connect to your remote slingbox. startup a viewing session with the sling software and grab that info from the Network Settings. Unless you really know for sure what there are.
On the router where the slingbox resides add a port forwarding rule to map the slingbox port to the internal IP address and port of the slingbox
In your config.ini file set the slingbox ipaddress and port number to the external ip address where the slingbox is located. 
Start the server and start watching.....

**Reply by fguardia** [reply]
Ok do i use the  public ip address or the home ip address

**Reply by GerryDazoo** [reply]
The public IP for the server configuration and the home ip for setting up the port forwarding rule on the remote router.

**Reply by htwonpoke** [reply]
I am in the same boat as fguardia. My slingbox is in another location. Been working on this off and on for a couple weeks now and continue to get same problem(Error connecting to  (IP X.X.X.X, Port). I am very new to all of this but im almost 100% sure I have my IP correct. Is it true that your slingbox will need to be port forwarded to work?

**Reply by rbf121** [reply]
Likely a port is already being forwarded if the slingplayer is working today.  I was in a similar situation since the slingbox I connected to is in another location.  I found the port being used by watching a slingbox.com stream from a web browser (Internet Explorer, if you still have that somewhere) and viewing "Settings", "Network Details" on the right side of screen.  The Public IP Address and Network Port is what you need.  If you don't have a static IP address then the public IP address may change.

**Reply by GerryDazoo** [reply]
htwonpoke. Yes, the port forwarding rule on the router where the slingbox resides in mandatory.
port_num -> local_ip:port_num

**Reply by htwonpoke** [reply]
Thanks for the info. Ive tried to startup a viewing session but im on a mac and it wont open. I always just watch my slingbox on iphone or firestick. 

I'm starting to think that my router is not correctly port forwarding. Everything I try just continues to show that it is closed. I successfully set up port forwarding at my current location.Which I think is necessary for the server setup, correct? Im going to try and reach out to the internet provider to see what im doing wrong with the port forwarding at the slingbox location.

**Reply by GerryDazoo** [reply]
You only need to configure port forwarding to the [SERVER] port, if you intend to acces the server form a remote location. In the server program you should see output that looks like

Checking for slingbox at  ('192.168.117.122', 5001)
('192.168.117.122', 5001) OK

Where the IP address and port number match the Public IP address you configured in your config.ini
If you see the OK then your port forwarding rule is OK.

Next time please reply with the server output, it will be helpful in figuring out why things aren't working,

**Reply by htwonpoke** [reply]
I made a lot of progress, can see video on VLC now! Appreciate the help! However,  I am running into issues with the video freezing but I think it’s because I’m on a Mac. I see another discussion talking about the same problem I’m having(video plays for about 30 sec then freezes but the audio continues to work). I’ll jump on that thread sometime to see if I can troubleshoot my video issues. 

Again, thanks for the help!

**Reply by GerryDazoo** [reply]
Do not use VLC on the Mac, there are known issues with it. 
see this discussion  https://github.com/GerryDazoo/Slinger/discussions/73

**Comment by theweatherguy** [comment]
Hello,

I’m completely lost with the instructions and I’m wondering if someone familiar with Macs can point me in the right direction.

I’d like to list the steps in the instructions and indicate what I don’t understand or what doesn’t seem to work.

_`Sudo- apt-get install python3-pip`_

	
I have python3 installed with HomeBrew and I can confirm that it is in my PATH.  

2. _copy "slingbox_server.py, config.ini and sling.service to /home/slingbox #_ 

	I have copied them to my home folder,  a user I created called slingbox in its home directory.
	But I don’t know what /home/slingbox # is.   I see a # sign In the directions  What is that?     Is the “#” a typo??  

3. _**sudo cp sling.service /etc/systemd/system/.**_

	I know what this command is trying to do, but there is no directory called /etc/systemd/system/ .     BTW, Is the “.” In this command a typo??

_4-6. sudo systemctl daemon-reload
   sudo systemctl enable sling.service
   sudo systemctl start sling.service_


	Of course, the above commands won’t work because I couldn't copy sling.service in /etc/systemd/system


I went through the config.ini file with a text editor.    I don’t understand the first three quarters of it.  It looks like a web page with multiple links.   Why are those links and stuff there??  

Well into the config.ini  file, I saw some areas that I guess I change to my internal IP address 192.168.2.58. and my port 5202

BUT I keep reading about “uncommenting” lines.  To comment a line, doesn’t it mean to put a # in front of the line?     If these lines have no “#” in front to begin with, how can I UNcomment the line??


Just for the heck of it , If I run a terminal command— python slingplayer_service.py

I immediately got an error message for something on line 9.

I would greatly appreciate any help.  Thanks in advance.   BTW, I'm trying to install on an iMac with Monterey.

**Reply by GerryDazoo** [reply]
Firstly, the info in the readme.txt was for setting up an a Raspberry Pi.
So to try to answer your questions.
Read the discussion about running the server on the Mac.
2) Yes the # was to indicate a following comment which doesn't exist remove it. /home/slingbox should be created when you make the slingbox user. It may not be in that location on the Mac
3) No that's not a typo. Cehck the Mac discussion. The service architecture is different.
Commenting out a line in config.ini means putting a ; at the start of the line. Many examples in the file
I see nothing in the config.ini that looks like web page stuff. Are you sure your looking at the right file
If when you run it and get the error right away please reply with the program output. I'm guessing you haven't installed all the required python modules per the readme. But Shouldn't you be running python3 not python?

**Comment by theweatherguy** [comment]
Thank you,Gerry, for your prompt response.  First, you explained what commenting out is.  I didn’t know that it was done with a semicolon (;) with this type of file.  (Thought it was a #) Yes there are many examples in the file. ( I guess you can figure out that I’m a newbie.)  

Somehow I thought the raspberry pi/Linux instructions could be used as a template for the Mac. Clearly not. I will recheck the Mac specific discussions for more guidance.  Maybe I’ve missed seeing them, but I haven’t noticed any simple step by steps for the Mac. It would be wonderful if a Mac person who has been successful with your code would be willing to post some basic Mac step by step instructions.

As for the html links, I’ll repost what I’m looking at in the config file when I get back in front of my computer.

As for the python3 command, I AM running the latest python3. Version 2.x is not installed. I have an alias “python” in my local/bin pointing to python3. Either command will execute python3 in my setup.

Thank you again. I hope I can get this working.

followup. The config file I downloaded was somehow not the right file. Sorry. I see that the actual config file is very straightforward.

**Comment by f520k510** [comment]
I get it working on a mac.  Here is the steps by steps.  
at the terminal type: sudo nano /etc/sysctl.conf and add these three lines, and then reboot
sysctl -w kern.ipc.maxsockbuf=8192000
sysctl -w net.inet.tcp.sendspace=7250000
sysctl -w net.inet.tcp.recvspace=7250000
(this is to increase the tcp buffer, you can try changing the two 7250000 to a bigger value but it works for me at 7250000)

Then install python for macos.  https://www.python.org/downloads/macos/
Then install modules netifaces and flask.

modify config.ini by adding your admin password, slingbox ip, and port.
commenting out line 361 and 865 on the slingbox_server.py by putting a # before the lines.
make sure port forward is working for your slingbox (this is mandatory)
run at terminal : python3 slingbox_server.py

view at http://server_ip:port/slingbox

**Reply by theweatherguy** [reply]
Thank you much, f520k510.   That's exactly what I needed to get going. I've learned a lot with this attempt to make this work so far.  I truly appreciate the generosity of GerryDazoo and people like you!

**Comment by theweatherguy** [comment]
Ok, I've gotten pretty far with the help, but when I enter the command     _python3 slingbox_server.py_   I get the following error showing on Terminal —

_Traceback (most recent call last):
  File "/Users/slingbox/slingbox_server.py", line 13, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'_



**Here is my current config file (I've altered my admin password and finderID to maintain privacy here)  *I have a slingbox 350 on port 5202 and ip 192.168.2.146
To get started, I'm NOT using netifaces or flask.  I just want to see if I can stream on my internal network 192.168.2.xx**



[SLINGBOX]
sbtype="350"
sbtype="Solo"
password=9TsYIURTLHoFrb
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. ALso removes requirement for the netifaces module
ipaddress=192.168.2.146
port=5202
; Repace with your finderid
finderid=89B8D779A13DF8XXXX86221E6VC141D43

name=DenSlingbox

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
FrameRate=30

;Min 50 Max 8000
VideoBandwidth=2000

; Min 0 (auto) max 63
VideoSmoothness=63

; Send Iframe every n seconds Max 30
IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corresponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, ???
Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't specify it will default to the last selected input
; So if you only have one video source connected you don't need to
; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter example
;VideoSource = 1

[SERVER]
; local port number for the server to listen on for connections
port=8080
maxstreams=3
enableremote=no

[REMOTE]
; see release notes for changing this if your remote doesn't work
;code=2
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
;include=remote.txt

Any help would be appreciated.

G

**Reply by GerryDazoo** [reply]
Your version of python is missing the requests module. This package isn't used anymore so you can safely comment out line 13 like you did for lines 361 and 865  per f520k510 instructions. ( put a # at the beginning of the line.

Two notes. You have two sbtypes defined in your config.ini
sbtype="350"
sbtype="Solo"

Delete the "Solo" line
Change sbtype="350" to sbtype="350/500/M1"  like it shows in the default config.ini

You don't need finederid defined for now. Comment out finderid
change finderid=89B8D779A13DF8XXXX86221E6VC141D43 to
;finderid=89B8D779A13DF8XXXX86221E6VC141D43

**Reply by theweatherguy** [reply]
Thank you very much, Gerry.  I will make those changes and report back.

**Comment by theweatherguy** [comment]
Hi Gerry,
Pleased to report I think I've made progress!  

I'm trying to connect with another Mac on the same network (ip address 192.168.2.75).   I tried VLC, although I know you said it's problematic. Then I tried with Firefox and I think that's what it's calling a "Hacker Alert"

(I'm assuming the IP address I put in to receive the **stream** is the IP address of the Mac server, not the slingbox.   Is that correct?)
The slingbox_ server.py is running on a Mac with IP address 192.168.2.165  
 
So to receive the stream, I'm using the URL  http://192.168.2.165:8080/DenSlingbox 

Not seeing any video, but it looks like the server is reporting an attempt to connect from Firefox as a "Hacker Alert"

I've made a lot of progress with your help.  I think I'm close to getting this to work. Maybe I need to find the proper Mac App to receive the stream?


Here's the terminal output:

_slingbox$ python3 slingbox_server.py
Version : 3.07 Running on macOS-12.6-x86_64-i386-64bit pid= 5698
Using config file config.ini
Connection Manager Running on port 8080 with 3 max streams....
Streamer Running:  3 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('192.168.2.146', 5201)
('192.168.2.146', 5201) OK
Using slingbox at  ('192.168.2.146', 5201)
Streamer:  DenSlingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Hacker Alert. Invalid Request from  ('192.168.2.75', 54062)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54063)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54064)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54065)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54066)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54067)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54068)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54069)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54070)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54071)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54094)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54095)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54096)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54097)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54098)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54099)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54100)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54101)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54102)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54103)
09/23/2022, 16:44:30   Streaming connection from ('192.168.2.75', 54129)

Slinginfo  "350/500/M1" 9TsYIkjfXzUoFEb 12 30 192.168.2.146 5201 3072
slingbox Starting Stream for  192.168.2.75:54129
Connecting... ('192.168.2.146', 5201) Control
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 32
Connecting... ('192.168.2.146', 5201) Stream
Hacker Alert. Invalid Request from  ('192.168.2.75', 54135)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54136)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54137)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54138)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54139)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54140)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54141)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54142)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54143)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54144)
slingbox Stream started at 09/23/2022, 16:44:33  744 0
09/23/2022, 16:44:33  slingbox Stream Terminated for  192.168.2.75:54129 [Errno 32] Broken pipe
slingbox Shutting down connections
Streamer:  slingbox Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Hacker Alert. Invalid Request from  ('192.168.2.75', 54158)
Hacker Alert. Invalid Request from  ('192.168.2.75', 54159)_

**Comment by GerryDazoo** [comment]
The proper url is
 http://192.168.2.165:8080/slingbox 
to stream. use http://192.168.2.165:8080/Remote for the web based remote control

**Comment by theweatherguy** [comment]
I thought, because I changed the name in the config file to be "DenSlingBox" that I needed that instead.  I'll change the name back in the config file to slingbox and I'll try the proper URL.

Thank you.   You've been great!

**Reply by theweatherguy** [reply]
Success- works like a charm on VLC for the iPad!    Thank you again, Gerry, for being the genius who created this beautiful hack and to everyone who else who helped me get it working.  I learned a lot!
 
My next step will be to get the channel changing to work.
When completed, I will take all the lessons learned and post a New discussion for an easy step-by-step for us computer newbies on the Mac.

**Comment by fguardia** [comment]
I am Stuck here
Version : 3.07 Running on Windows-10-10.0.22000-SP0 pid= 21360
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams....
Streamer Running:  10 config.ini 8080
starting up on  port 8080
Checking for slingbox at ('170.250.69.39', 5205)
('170.250.69.39', 5205) OK
Using slingbox at  ('170.250.69.39', 5205)
Streamer:   Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses (0.0.0.0)
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://127.0.0.1:8081
 * Running on http://192.168.128.224:8081 (Press CTRL+C to quit)
09/29/2022, 07:46:20  Registering Slingboxes {'8DF996E5D69D9B7E1DBF4156FDDB53FD': 8080}
Getting external IP address
Redirector, my external IP address 190.14.210.6
PING failure ('190.14.210.6', 8080) [WinError 10060] Se produjo un error durante el intento de conexión ya que la parte conectada no respondió adecuadamente tras un periodo de tiempo, o bien se produjo un error en la conexión establecida ya que el host conectado no ha podido responder
No working external portmaps. Not registering slingboxes
Retrying in 10 minutes

**Reply by fguardia** [reply]
my config looks like this
[SLINGBOX]
sbtype="350/500/M1"
;sbtype="Solo/Pro/ProHD"
password=q6ZitX9prIUDr4L
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your 
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. ALso removes requirement for the netifaces module  
ipaddress=170.250.69.39
port=5205
; Repace with your finderid
finderid=8DF996E5D69D9B7E1DBF4156FDDB53FD

;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
;VideoBandwidth=2000

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, ???
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't specify it will default to the last selected input
; So if you only have one video source connected you don't need to 
; mess with this. This number "might" be the same as the [REMOTE] code= entry but I can't say for sure. Please ket me know if you find a counter example
;VideoSource = 1

[SERVER]
; local port number for the server to listen on for connections
port=8080
maxstreams=10
enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work
;code=2
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
;include=remote.txt

**Reply by GerryDazoo** [reply]
Everything looks good to me. Please comment out the finderid line to get rid of those PING failures. You don't have the proper router port forwarding rules in place.

So I guess my question is "What isn't working?"
I see no streaming requests in the program output  http://192.168.128.224:8080/slingbox
If you are trying to stream and you don't see output from the program, then I would suspect you local firewall is blocking the requests.

**Reply by tom76017** [reply]
The ip address, 170.250.69.39, looks like a Public IP, not a LAN ip.  You need to get your local ip from your router or the HOME IP Address from the Slingplayer Desktop

**Reply by fguardia** [reply]
I tried the home ip and got the same result gonna try to test it when i go back to where the slingbox is next month
